addons/portal/controllers/mail.py

```diff
@@ -117,11 +22,19 @@ def _portal_post_has_content(self, thread_model, thread_id, message, attachment_
 
     @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
     def portal_avatar(self, res_id=None, height=50, width=50, access_token=None, _hash=None, pid=None):
-        """ Get the avatar image in the chatter of the portal """
+        """Get the avatar image in the chatter of the portal"""
         if access_token or (_hash and pid):
-            message = request.env['mail.message'].browse(int(res_id)).exists().filtered(
-                lambda msg: _check_special_access(msg.model, msg.res_id, access_token, _hash, pid and int(pid))
-            ).sudo()
+            message = (
+                request.env["mail.message"]
+                .browse(int(res_id))
```

Comment:
This is quite weird ... you browse an int, call exists (ok it filters sudo-like as it is done in sql), then you filter (using msg fields hence message access is required), then you sudo it ? There are unclear stuff to me, what is the expected security check and input/output here ?

Reply:
The logic is as before. If a user has access to a thread, They can see the avatars of the messages. The only difference is checking access with `_check_special_access()` is replaced by `_get_thread_with_access()`.

Reply:
In general, if it makes the coce more readable `browse exists sudo` can be replaced by `sudo (active-test=False if model has the field) search`.
In this case as there's no active field, it's probably going to be cleaner with just `sudo search`

Reply:
But I guess the `sudo` is last here to not mess up with `_get_thread_with_access`, so maybe there is no better way.
Unless we ensure `_get_thread_with_access` removes sudo internally.

Reply:
I think this sudo after checking access is actually useless (In both master and here). In checking access, we're already using the message fields.